### PR TITLE
Milestone 9: add devise

### PR DESCRIPTION
**Pull request by ITurres**
## Pull Request Summary for Milestone 9 Completion

- [x] All [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#project-requirements) completed.

---

#### `/app/devise/*`

- @ITurres && @demesameneshoa: We have added the `devise` gem to the project.
- We have modified the usual devise files to fit the project needs, that includes:

  - modified views. (Updated: now with `confirmable`)
  - modified partials. (Updated: now with `confirmable`)
  - modified controllers.
  - modified environments.
  - modified routes.
  - modified models. (Updated: now with `confirmable`)
  - Added migrations. (Updated: now with `confirmable`)
  - Regenerate schema. (Updated: now with `confirmable`)

  [Please see here for further details](https://github.com/ITurres/blog_app/pull/9/files)

#### `/app/views/layouts/_navbar.html.erb`

- @ITurres: I have added a navbar partial to display the app logo and the sign out link as long as the user is logged in.

#### `/app/views/layouts/_back_button.html.erb`

- @ITurres: I have added a back button partial to display a back button as long as the current page is not root.

---

#### `/app/models/user.rb`

- @ITurres: Apart form the devise modifications, I have added a private `set_default_photo` method to set a default random photo to the user before creation.

- Marked the `devise` task as completed.

#### `/app/views/layouts/application.html.erb`

- @ITurres: now the `navbar` partial is being rendered, and the `back_button`.

#### `/app/views/shared/\_user_card.html.erb

- @ITurres: I have modified/added bootstrap classes to the user card.

---

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (Shedrack-Sunday) [inline app/models/user.rb]: - [x] Kindly enable `:confirmable` to make it possible to get a confirmation after login 👍 

### Requirement:
image

File: db/migrate/20231213164642_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (Shedrack-Sunday) [inline db/migrate/20231213164642_add_devise_to_users.rb]: - [x] Do well to uncomment this section, to make sure the confirmable configuration works 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @ITurres     👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with  Devise features ✔️
- Great work on the  Authorization implementation 👏   



# - Required Changes  🌵  

- [x] Kindly note  the comment i made regarding the issues on enabling the `:onfirmable` feature to make sure signup is confirmed 
# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by ITurres** [comment]
Holly Molly, we have overlooked that @demesameneshoa ! 🥲

Hi @Shedrack-Sunday! thank you so much for having reviewed this PR! You are right!, We'll fix it right away 🎯
We'll also make sure to modify the views to math the current design.

Thank you again! 🥂

**Review by V-Blaze — APPROVED** [review]
Hi @ITurres, @demesameneshoa  

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Devise setup correctly 💯
-  Great communication with reviewer👍
-  Correct git flow🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ITurres** [comment]
Wow great! 🥳. Thank you @V-Blaze! We both wish you a great week! thank you for reviewing the PR.

**Pull request merged**